wooey/views/views.py

```diff
@@ -65,23 +76,78 @@ def post(self, request, *args, **kwargs):
                 if valid is True and group_valid is True:
                     job = utils.create_wooey_job(script_pk=script_pk, user=user, data=form.cleaned_data)
                     job.submit_to_celery()
-                    return JsonResponse({'valid': True})
-            return JsonResponse({'valid': False, 'errors': {'__all__': [force_text(_('You are not permitted to access this script.'))]}})
-        return JsonResponse({'valid': False, 'errors': form.errors})
+                    return {'valid': True, 'job_id': job.id}
+
+            return {'valid': False, 'errors': {'__all__': [force_text(_('You are not permitted to access this script.'))]}}
+
+        return {'valid': False, 'errors': form.errors}
+
+
+class WooeyScriptJSON(WooeyScriptBase):
+
+    # FIXME: the form data is returned as form objects so can be passed to templates
+    # this render_fn allows us to pass the return through a stringify method for JSON
+    render_fn = lambda form: form.as_table()
+
+    def render_to_response(self, context, *args, **kwargs):
+        return JsonResponse(context)
+
+    def post(self, *args, **kwargs):
+        data = super(WooeyScriptJSON, self).post(*args, **kwargs)
+        return JsonResponse(data)
+
+class WooeyScriptView(WooeyScriptBase):
+
+    template_name = 'wooey/scripts/script_view.html'
+
+    def post(self, *args, **kwargs):
+        data = super(WooeyScriptView, self).post(*args, **kwargs)
+        if data['valid']:
+            return HttpResponseRedirect( reverse('wooey:celery_results', kwargs={'job_id': data['job_id'] }) )
+        else:
+            # FIXME: This works but the form handling here should return the submitted data
+            # may need to refactor the JSON stuff a little bit to make this work
+            return self.get(*args, **kwargs)
+
 
 
 class WooeyHomeView(TemplateView):
-    template_name = 'wooey/wooey_home.html'
+    template_name = 'wooey/home.html'
 
     def get_context_data(self, **kwargs):
-        job_id = self.request.GET.get('job_id')
+        #job_id = self.request.GET.get('job_id')
         ctx = super(WooeyHomeView, self).get_context_data(**kwargs)
-        ctx['wooey_scripts'] = getattr(settings, 'WOOEY_SCRIPTS', {})
-        if job_id:
-            job = WooeyJob.objects.get(pk=job_id)
-            if job.user is None or (self.request.user.is_authenticated() and job.user == self.request.user):
-                ctx['clone_job'] = {'job_id': job_id, 'url': job.get_resubmit_url(), 'data_url': job.script.get_url()}
+        ctx['scripts'] = Script.objects.all()
+
+
+        # Get the id of every favorite (scrapbook) file
+        ctype = ContentType.objects.get_for_model(Script)
+        ctx['favorite_script_ids'] = Favorite.objects.filter(content_type=ctype, user=self.request.user).values_list('object_id', flat=True)
```

Comment:
This will fail if the user is Anonymous. We need to add a utility function for fetching users that will 1) evaluate self.request.user to make sure it isn't Lazy, 2) check if we are permitting anonymous users, 3) return an appropriate user for the anonymous case.